sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -980,15 +980,25 @@ object EliminateSorts extends Rule[LogicalPlan] {
       if (newOrders.isEmpty) child else s.copy(order = newOrders)
     case Sort(orders, true, child) if SortOrder.orderingSatisfies(child.outputOrdering, orders) =>
       child
-    case s @ Sort(_, _, child) => s.copy(child = recursiveRemoveSort(child))
+    case s @ Sort(_, _, child) => s.copy(child = recursiveRemoveSort(s, child))
     case j @ Join(originLeft, originRight, _, cond, _) if cond.forall(_.deterministic) =>
       j.copy(left = recursiveRemoveSort(originLeft), right = recursiveRemoveSort(originRight))
     case g @ Aggregate(_, aggs, originChild) if isOrderIrrelevantAggs(aggs) =>
       g.copy(child = recursiveRemoveSort(originChild))
   }
 
+  private def recursiveRemoveSort(parent: Sort, plan: LogicalPlan): LogicalPlan = plan match {
+    case s @ Sort(_, _, child) =>
+      if (parent.global == s.global) {
```

Comment:
How about the case , `global sort => project/filter => local sort`?

Reply:
I add new UT for this.